src/app/clusters/ota-provider/ota-provider-delegate.h

```diff
@@ -35,15 +37,12 @@ namespace Clusters {
 class OTAProviderDelegate
 {
 public:
-    // TODO(#8605): protocolsSupported should be list of OTADownloadProtocol enums, not uint8_t
-    virtual EmberAfStatus HandleQueryImage(CommandHandler * commandObj, const ConcreteCommandPath & commandPath, uint16_t vendorId,
-                                           uint16_t productId, uint32_t softwareVersion, uint8_t protocolsSupported,
-                                           const Optional<uint16_t> & hardwareVersion, const Optional<CharSpan> & location,
-                                           const Optional<bool> & requestorCanConsent,
-                                           const Optional<ByteSpan> & metadataForProvider) = 0;
-
-    virtual EmberAfStatus HandleApplyUpdateRequest(CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-                                                   const chip::ByteSpan & updateToken, uint32_t newVersion) = 0;
+    virtual EmberAfStatus HandleQueryImage(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                           const OtaSoftwareUpdateProvider::Commands::QueryImage::DecodableType & commandData) = 0;
+
+    virtual EmberAfStatus
+    HandleApplyUpdateRequest(CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+                             const OtaSoftwareUpdateProvider::Commands::ApplyUpdateRequest::DecodableType & commandData) = 0;
 
     virtual EmberAfStatus HandleNotifyUpdateApplied(const chip::ByteSpan & updateToken, uint32_t softwareVersion) = 0;
```

Comment:
Might as well be consistent and make the same change to `HandleNotifyUpdateApplied`, no?

Reply:
Done.